src/Compilers/CSharp/Portable/Binder/Imports.cs

```diff
@@ -217,6 +217,8 @@ internal string GetDebuggerDisplay()
                                 }
                                 else
                                 {
+                                    Binder.ReportDiagnosticsIfObsoleteInternal(diagnostics, importedType, usingDirective.Name, importedType.ContainingSymbol, BinderFlags.None);
```

Comment:
>Binder.ReportDiagnosticsIfObsoleteInternal [](start = 36, length = 42)

It doesn't look like we are supposed to use this method directly. I would use ```internal void ReportDiagnosticsIfObsolete(DiagnosticBag diagnostics, Symbol symbol, SyntaxNodeOrToken node, bool hasBaseReceiver)``` and call it off of ``declarationBinder``` or ```usingsBinder```. Might be useful to check how VB does this.